lib/ReportDataCollector.php

```diff
@@ -269,46 +245,36 @@ private function hideMountPasswords(IStorageConfig $mount, array &$configArray)
 					$authParameters[$key]->getType() === \OCP\Files\External\DefinitionParameter::VALUE_PASSWORD
 				)
 			) {
-				$configArray[$key] = \OCP\IConfig::SENSITIVE_VALUE;
+				$configArray[$key] = IConfig::SENSITIVE_VALUE;
 			}
 		}
 	}
 
-	/**
-	 * @return array
-	 */
-	private function getIntegrityCheckerDetailArray() {
+	private function getIntegrityCheckerDetailArray(): array {
 		return [
 			'passing' => $this->integrityChecker->hasPassedCheck(),
 			'enabled' => $this->integrityChecker->isCodeCheckEnforced(),
 			'result' => $this->integrityChecker->getResults(),
 		];
 	}
 
-	/**
-	 * Basic report data
-	 * @return array
-	 */
-	private function getBasicDetailArray() {
+	private function getBasicDetailArray(string $licenseKey): array {
 		return [
-			'license key' => $this->licenseKey,
+			'license key' => $licenseKey,
```

Comment:
maybe I'm missing something, but I don't see the value change... why bother to include a "license key" field when it will always contain `IConfig::SENSITIVE_VALUE`?

Reply:
the license key is transmitted via the http internface. It is necessary as some kind of identifier for customers.